frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/affinity-modal/affinity-modal.tsx

```diff
@@ -183,17 +229,33 @@ export const AffinityModal = withHandlePromise<AffinityModalProps>(
                 </div>
               )}
               {affinities.length > 0 ? (
-                <AffinityTable
-                  columnClasses={columnClasses}
-                  data={affinities}
-                  customData={{
-                    isDisabled: false,
-                    vmLikeFinal,
-                    onEdit: onAffinityClickEdit,
-                    onDelete: onAffinityDelete,
-                  }}
-                  row={AffinityRow}
-                />
+                <Stack>
+                  <StackItem>
+                    <AffinityTable
+                      columnClasses={columnClasses}
+                      data={affinities}
+                      customData={{
+                        isDisabled: false,
+                        vmLikeFinal,
+                        onEdit: onAffinityClickEdit,
+                        onDelete: onAffinityDelete,
+                      }}
+                      row={AffinityRow}
+                    />
+                  </StackItem>
+                  {requiredNodeAffinities?.length > 0 && (
```

Comment:
so if you have empty required nodes we don't show the matching node table. That is a bit confusing since the VM will schedule alright. Shouldn't we default to all nodes in that case?